src/coreclr/inc/cordebug.idl

```diff
@@ -3988,14 +3988,14 @@ interface ICorDebugRegisterSet : IUnknown
         REGISTER_LOONGARCH64_F31,
 
         REGISTER_RISCV64_PC = 0,
-        REGISTER_RISCV64_RA,
         REGISTER_RISCV64_SP,
+        REGISTER_RISCV64_FP,
+        REGISTER_RISCV64_RA,
         REGISTER_RISCV64_GP,
```

Comment:
There is `REGISTER_RISCV64_X0` with TODO at the end of this list that does not appear to be used anywhere. Delete it while you are on it?

Reply:
I am still not sure about `REGISTER_RISCV64_X0`, plus, I see it used here for some purpose:
https://github.com/dotnet/runtime/blob/23695f9897ed5d0b4004db91a3f3835fc26b7a3c/src/coreclr/debug/inc/riscv64/primitives.h#L52

Note, current implementation aimed and tested only with diagnostics SOS command with simple tests, that is extremely limited and I belive not cover all cases of this code usage. So, I leave it as is for now.

Reply:
I think `REGISTER_RISCV64_X0` tries to be the R0 register that is not a real register, so it does not need to be in the enum.

Reply:
@clamp03 could you please clarify, is X0 here used in the R0 meaning?

Reply:
`ICorDebugInfo::RegNum` have `REGNUM_R0` https://github.com/dotnet/runtime/blob/0ad3723332c37f6423a725fbd4ad889b8e0f3d7c/src/coreclr/inc/cordebuginfo.h#L183
and `REGISTER_RISCV64_X0` used just as "padding" for proper mapping with `g_JITToCorDbgReg`
https://github.com/dotnet/runtime/blob/23695f9897ed5d0b4004db91a3f3835fc26b7a3c/src/coreclr/debug/inc/riscv64/primitives.h#L155-L165

I am not really strong in this code, do we really could remove R0 from `ICorDebugInfo::RegNum` ? Is not  `ICorDebugInfo::RegNum` used as indexes for "normal" (not debug related) register context?

Reply:
This padding is an internal implementation detail. It should not leak through the public ICorDebug surface.

Arm64 has similar wzr register and we do not have a constant for it in the ICorDebug register enum either.

Reply:
Thank you for explanation, I got the idea. Will prepare appropriate changes.